lib/lib_ssl/bearssl-esp8266/src/ssl/ssl_ccert_single_ec.c

```diff
@@ -59,7 +59,7 @@ cc_choose(const br_ssl_client_certificate_class **pctx,
 	scurve = br_ssl_client_get_server_curve(cc);
 
 	if ((zc->allowed_usages & BR_KEYTYPE_KEYX) != 0
-		&& scurve == zc->sk->curve)
+		&& zc->sk && scurve == zc->sk->curve)
```

Comment:
You shouldn't need to change the underlying Bearssl library. Otherwise it should be reported upstream. But I doubt the problem is in the library.

Reply:
The problem is you can get till here with sk == NULL. The problem may be the call below but read there other note

Reply:
However this is a bad "library" behaviour, it's not wise to not validate variables

Reply:
Yes and no. You are not supposed to pass a null pointers anyways for your certificate